snowpack/src/build/build-import-proxy.ts

```diff
@@ -1,33 +1,25 @@
 import type CSSModuleLoader from 'css-modules-loader-core';
 import path from 'path';
 import {SnowpackConfig} from '../types/snowpack';
-import {getExt, URL_HAS_PROTOCOL_REGEX} from '../util';
+import {getExt} from '../util';
 import {logger} from '../logger';
 
 const CLOSING_BODY_TAG = /<\s*\/\s*body\s*>/gi;
 
-export function getMetaUrlPath(urlPath: string, isDev: boolean, config: SnowpackConfig): string {
-  let {baseUrl, metaDir} = config.buildOptions || {};
-  if (isDev) {
-    return path.posix.normalize(path.posix.join('/', metaDir, urlPath));
-  }
-  if (URL_HAS_PROTOCOL_REGEX.test(baseUrl)) {
-    return baseUrl + path.posix.normalize(path.posix.join(metaDir, urlPath));
-  }
-  return path.posix.normalize(path.posix.join(baseUrl, metaDir, urlPath));
+export function getMetaUrlPath(urlPath: string, config: SnowpackConfig): string {
+  let {metaDir} = config.buildOptions || {};
+  return path.posix.normalize(path.posix.join('/', metaDir, urlPath));
```

Comment:
okay, this was the real fix that was needed: `baseUrl` is unnecessary, since it's added automatically when we go to actually resolve these URLs.  This is meant to be a pre-resolved URL, baseUrl is added later.

Reply:
The nice part is that this means no need for different behavior between dev and build. `isDev` is no longer needed

Reply:
Ah makes sense. I think I realized a parallel truth in the optimize PR, and that’s the realization that we can deal only in absolute (local) filesystem URLs right up until it’s time to write that file back out (or in the dev server’s case, sending a file).

+1 for deferring things like `baseUrl` until the time we send a file (it’s basically just replacing the first segment of a file as you‘re outputting).